tangelo/toolboxes/post_processing/post_selection.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2022 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module provides functions to create symmetry verification and post-selection circuits."""
+
+import warnings
+
+from tangelo.linq import Circuit, Gate
+from tangelo.linq.helpers import measurement_basis_gates, pauli_string_to_of
+from tangelo.toolboxes.operators import QubitOperator, count_qubits
+from tangelo.toolboxes.post_processing import Histogram
+
+
+def ancilla_symmetry_circuit(circuit, sym_op):
+    """Append a symmetry operator circuit to an input circuit using an extra ancilla qubit,
+    for the purpose of symmetry verification and post-selection.
+    For more details see arXiv:1807.10050.
+
+    Args:
+        circuit (Circuit): A quantum circuit to be equipped with symmetry verification
+        sym_op (string): The symmetry operator to be verified.
+            Can be a Pauli string, OpenFermion style list or a QubitOperator
+
+    Returns:
+        Circuit: The input circuit appended with the proper basis rotation
+            and entanglement with an ancilla qubit, which is added as the last qubit.
+            Increases the input circuit width by 1."""
+    if isinstance(sym_op, QubitOperator):
+        op_len = count_qubits(sym_op)
+    else:
+        op_len = len(sym_op)
+    n_qubits = circuit.width
+
+    # Check if the operator size matches the circuit width
+    if n_qubits < op_len:
+        raise RuntimeError("The size of the symmetry operator is bigger than the circuit width.")
+    elif n_qubits > op_len:
+        warnings.warn("The size of the symmetry operator is smaller than the circuit width. Remaining qubits will be measured in the Z-basis.")
+
+    if isinstance(sym_op, str):
```

Comment:
You have no `else` here. I don't think it's something you've done in this function: if the `symptoms_op` is none of these 3 things then we should return an error. That should cover our backs if the user pushes in something of the wrong type.

Reply:
Added error message